cmd/src/campaigns_apply.go

```diff
@@ -0,0 +1,233 @@
+		}
+
+		opts := campaigns.ExecutorOpts{
+			Cache:      svc.NewExecutionCache(*cacheDirFlag),
+			ClearCache: *clearCacheFlag,
+			KeepLogs:   *keepFlag,
+			Timeout:    *timeoutFlag,
+		}
+		if parallelismFlag != nil || *parallelismFlag <= 0 {
+			opts.Parallelism = runtime.GOMAXPROCS(0)
+		} else {
+			opts.Parallelism = *parallelismFlag
+		}
+		executor := svc.NewExecutor(opts, nil)
+
+		if errs != nil {
+			return errs
+		}
+
+		pending := createPending(out, "Parsing campaign spec")
+		campaignSpec, err := svc.ParseCampaignSpec(specFile)
+		if err != nil {
+			return errors.Wrap(err, "parsing campaign spec")
+		}
+
+		if err := campaignsValidateSpec(out, campaignSpec); err != nil {
+			return err
+		}
+		completePending(pending, "Parsing campaign spec")
+
+		pending = createPending(out, "Resolving namespace")
+		namespace, err := svc.ResolveNamespace(ctx, *namespaceFlag)
+		if err != nil {
+			return err
+		}
+		completePending(pending, "Resolving namespace")
+
+		var progress output.Progress
+		specs, err := svc.ExecuteCampaignSpec(ctx, executor, campaignSpec, func(statuses []*campaigns.TaskStatus) {
+			if progress == nil {
+				progress = out.Progress([]output.ProgressBar{{
+					Label: "Executing campaign spec",
+					Max:   float64(len(statuses)),
+				}}, nil)
+			}
+
+			complete := 0
+			for _, ts := range statuses {
+				if !ts.FinishedAt.IsZero() {
+					complete += 1
+				}
+			}
+			progress.SetValue(0, float64(complete))
+		})
+		if err != nil {
+			return err
+		}
+		if progress != nil {
+			progress.Complete()
+		}
+
+		if logFiles := executor.LogFiles(); len(logFiles) > 0 && *keepFlag {
+			block := out.Block(output.Line("", successColor, "Preserving log files:"))
+			for _, file := range logFiles {
+				block.Write(file)
+			}
+		}
+
+		progress = out.Progress([]output.ProgressBar{
+			{Label: "Sending changeset specs", Max: float64(len(specs))},
+		}, nil)
+		ids := make([]campaigns.ChangesetSpecID, len(specs))
+		for i, spec := range specs {
+			id, err := svc.CreateChangesetSpec(ctx, spec)
+			if err != nil {
+				return err
+			}
+			ids[i] = id
+			progress.SetValue(0, float64(i+1))
+		}
+		progress.Complete()
+
+		pending = createPending(out, "Creating campaign spec on Sourcegraph")
+		id, url, err := svc.CreateCampaignSpec(ctx, namespace, campaignSpec, ids)
+		if err != nil {
+			return err
+		}
+		completePending(pending, "Creating campaign spec on Sourcegraph")
+
+		if *applyFlag {
+			pending := createPending(out, "Applying campaign")
+			campaign, err := svc.ApplyCampaign(ctx, id)
+			if err != nil {
+				return err
+			}
+			completePending(pending, "Applying campaign")
+
+			out.Write("")
+			block := out.Block(output.Line(successEmoji, successColor, "Campaign applied!"))
+			block.Write("To view the campaign, go to:")
+			block.Writef("%s%s", cfg.Endpoint, campaign.URL)
+		} else {
+			out.Write("")
+			block := out.Block(output.Line(successEmoji, successColor, "To apply the campaign, go to:"))
```

Comment:
Of those two options, I prefer my wording on this, since it's a clearer call to action, but let me see if I can come up with some wording somewhere between the two.